contracts/yield-source/YearnV2YieldSource.sol

```diff
@@ -164,7 +180,11 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable {
         uint256 yShares = _tokenToYShares(amount);
         uint256 previousBalance = token.balanceOf(address(this));
         // we accept losses to avoid being locked in the Vault (if losses happened for some reason)
-        vault.withdraw(yShares, address(this), MAX_LOSSES);
+        if(maxLosses != 0) {
+            vault.withdraw(yShares, address(this), maxLosses);
```

Comment:
It would be good to have this covered in testing.  Just need to set max loss then attempt a deposit

Reply:
done